core/domain/skill_services.py

```diff
@@ -230,6 +232,167 @@ def get_all_skill_summaries():
+
+    Args:
+        status: string. The status of the skill.
+        classroom_name: string. The classroom_name of the topic
+            to which skill is assigned to.
+        keywords: list(string). The keywords to look for
+            in the skill description.
+        sort_by: string. A string indicating how to sort the result.
+
+    Returns:
+        list(dict). The list of skill summary dicts matching the filters.
+    """
+    skill_summary_dicts = get_all_skill_summary_dicts_with_topic_and_classroom()
+
+    filtered_skill_summary_dicts = filter_skills_by_status(
+        skill_summary_dicts, status)
+    filtered_skill_summary_dicts = filter_skills_by_classroom(
+        filtered_skill_summary_dicts, classroom_name)
+    filtered_skill_summary_dicts = filter_skills_by_keywords(
+        filtered_skill_summary_dicts, keywords)
+
+    if sort_by == (
+            constants.TOPIC_SKILL_DASHBOARD_SORT_OPTIONS['DecreasingCreatedOn']):  # pylint: disable=line-too-long
+        return sorted(
+            filtered_skill_summary_dicts,
+            key=lambda x: x['skill_model_created_on'], reverse=True)
+    elif sort_by == (
+            constants.TOPIC_SKILL_DASHBOARD_SORT_OPTIONS['IncreasingUpdatedOn']):  # pylint: disable=line-too-long
+        return sorted(
+            filtered_skill_summary_dicts,
+            key=lambda x: x['skill_model_last_updated'])
+    elif sort_by == (
+            constants.TOPIC_SKILL_DASHBOARD_SORT_OPTIONS['DecreasingUpdatedOn']):  # pylint: disable=line-too-long
+        return sorted(
+            filtered_skill_summary_dicts,
+            key=lambda x: x['skill_model_last_updated'], reverse=True)
+    else:
+        return sorted(
+            filtered_skill_summary_dicts,
+            key=lambda x: x['skill_model_created_on'])
+
+
+def get_all_skill_summary_dicts_with_topic_and_classroom():
+    """Returns all the skill summary dicts after attaching topic and classroom.
+
+    Returns:
+        list(dict). The list of skill summary dicts matching the filters.
+    """
+    skill_summaries = get_all_skill_summaries()
+    total_skill_summaries = [summary.to_dict() for summary in skill_summaries]
+
+    skill_ids_assigned_to_some_topic_with_topic_details = (
+        topic_services.get_all_skill_ids_assigned_to_some_topic_with_topic_details())  # pylint: disable=line-too-long
+
+    for skill_summary_dict in total_skill_summaries:
+        skill_summary_dict['topic'] = None
+        skill_summary_dict['classroom'] = None
+        for assigned_skill in skill_ids_assigned_to_some_topic_with_topic_details:  # pylint: disable=line-too-long
+            if skill_summary_dict['id'] == assigned_skill[0]:
+                skill_summary_dict['topic'] = assigned_skill[1]
+                skill_summary_dict['classroom'] = assigned_skill[2]
+                break
+    return total_skill_summaries
+
+
+def filter_skills_by_status(skill_summary_dicts, status):
```

Comment:
Here and elsewhere, you should be working with domain objects, not dicts.

Reply:
Done.